de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolRequest/presenter/ToolRequestPresenterImpl.java

```diff
@@ -122,12 +121,7 @@ public void setViewDebugId(String baseId) {
 
     @Override
     public void onAdminMakeToolPublicSelected(AdminMakeToolPublicSelectedEvent event) {
-        adminView = adminFactory.create(new ListStore<>(new ModelKeyProvider<Tool>() {
-            @Override
-            public String getKey(Tool item) {
-                return item.getId();
-            }
-        }));
+        adminView = adminFactory.create(new ListStore<>(item -> item.getId()));
```

Comment:
This is really clean!

Mostly because I don't know how Java works (lol): how does this syntax know to override the "getKey" method specifically, or does it just do some sort of different handling when it's passed a function rather than a class, or what?